homeassistant/components/sfr_box/sensor.py

```diff
@@ -0,0 +1,182 @@
+        key="noise_down",
+        name="Noise down",
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS,
+        state_class=SensorStateClass.MEASUREMENT,
+        has_entity_name=True,
+        value_fn=lambda x: x.noise_down,
+    ),
+    SFRBoxSensorEntityDescription(
+        key="noise_up",
+        name="Noise up",
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS,
+        state_class=SensorStateClass.MEASUREMENT,
+        has_entity_name=True,
+        value_fn=lambda x: x.noise_up,
+    ),
+    SFRBoxSensorEntityDescription(
+        key="attenuation_down",
+        name="Attenuation down",
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS,
+        state_class=SensorStateClass.MEASUREMENT,
+        has_entity_name=True,
+        value_fn=lambda x: x.attenuation_down,
+    ),
+    SFRBoxSensorEntityDescription(
+        key="attenuation_up",
+        name="Attenuation up",
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS,
+        state_class=SensorStateClass.MEASUREMENT,
+        has_entity_name=True,
+        value_fn=lambda x: x.attenuation_up,
+    ),
+    SFRBoxSensorEntityDescription(
+        key="rate_down",
+        name="Rate down",
+        device_class=SensorDeviceClass.DATA_RATE,
+        native_unit_of_measurement=UnitOfDataRate.KILOBITS_PER_SECOND,
+        state_class=SensorStateClass.MEASUREMENT,
+        has_entity_name=True,
+        value_fn=lambda x: x.rate_down,
+    ),
+    SFRBoxSensorEntityDescription(
+        key="rate_up",
+        name="Rate up",
+        device_class=SensorDeviceClass.DATA_RATE,
+        native_unit_of_measurement=UnitOfDataRate.KILOBITS_PER_SECOND,
+        state_class=SensorStateClass.MEASUREMENT,
+        has_entity_name=True,
+        value_fn=lambda x: x.rate_up,
+    ),
+    SFRBoxSensorEntityDescription(
+        key="line_status",
+        name="Line status",
+        device_class=SensorDeviceClass.ENUM,
+        options=[
+            "No Defect",
+            "Of Frame",
+            "Loss Of Signal",
+            "Loss Of Power",
+            "Loss Of Signal Quality",
+            "Unknown",
+        ],
+        has_entity_name=True,
+        value_fn=lambda x: x.line_status,
+    ),
+    SFRBoxSensorEntityDescription(
+        key="training",
+        name="Training",
+        device_class=SensorDeviceClass.ENUM,
+        options=[
+            "Idle",
+            "G.994 Training",
+            "G.992 Started",
+            "G.922 Channel Analysis",
+            "G.992 Message Exchange",
+            "G.993 Started",
+            "G.993 Channel Analysis",
+            "G.993 Message Exchange",
+            "Showtime",
+            "Unknown",
+        ],
+        has_entity_name=True,
```

Comment:
`has_entity_name` is set to  true on all of them... maybe just set it as a class variable instead.

Reply:
Addressed in #84977